syntax = "proto3";
import "google/protobuf/timestamp.proto";

package info.v1;

/* Device represents a physical microcontroller device */

message Device {

    // todo: add get/reply-type messages?

    /* semver for BlinkaConnect library version */
    message Version {
        uint64 major = 1;
        uint64 minor = 2;
        uint64 micro = 3;
        string label = 4;
      }
    Version version = 1;


    string unique_id = 2;
    string hardware_id = 3;
    google.protobuf.Timestamp timestamp = 4;
    float bat_level = 5;

    message Location {
        float lat = 1;
        float lon = 2;
        float ele = 3;
      }
    Location location = 6;
}